src/proto/uprotocol/uauthority.rs

```diff
@@ -65,4 +78,51 @@ impl UAuthority {
         self.remote = Some(Remote::Id(id));
         self
     }
+
+    /// Returns whether the `Remote` `Ip` meets conformance with IPv4 or IPv6 spec
+    ///
+    /// # Returns
+    /// Returns a `Result<IpConformance, ValidationError>` where the error means remote is not IP or there is no remote
+    /// and happy path tells us whether the IP conforms with spec or not
+    ///
+    /// # Errors
+    ///
+    /// Returns a `ValidationError` in the failure case, indicating no remote or remote is not IP
+    pub fn remote_ip_conforms(&self) -> Result<IpConformance, ValidationError> {
+        if let Some(_remote) = self.remote.as_ref() {
+            match &self.remote {
+                Some(Remote::Ip(ip)) => Ok(match ip.len() {
+                    REMOTE_IPV4_BYTES => IpConformance::IPv4,
+                    REMOTE_IPV6_BYTES => IpConformance::IPv6,
+                    _ => IpConformance::NonConformal,
+                }),
+                _ => Err(ValidationError::new("Remote is not IP")),
+            }
+        } else {
+            Err(ValidationError::new("No remote"))
```

Comment:
Is this actually an error? If I want to invoke a local service then there is no *remote*, or is there?

Reply:
So my thought process based on how things were laid out atm, was that in the `UriValidator::validate_micro_form()` we'd walk thru the logic in a way s.t. `remote_ip_conforms()` is only called if we know there's a remote.

Based on some of your other feedback, [like your overall review](https://github.com/eclipse-uprotocol/uprotocol-rust/pull/18#pullrequestreview-1814829206) it seems you have other thoughts.

Still gotta meditate more on your overall review to make sure I get it.